### ISSUE TITLE
Do not allow access to cleartext credentials from network

### ISSUE THREAD

**Issue by schlamar** [jomjol/AI-on-the-edge-device#3216]
### Issue

Modern IT security is all about layered security. Make each boundary of every application / device as robust as possible. One of the important paradigms is always assume there is an attacker in the network.

One worst case scenario in this context is getting access to credentials (Wifi, MQTT, ...) via an unauthorized network communication. 

So I'm not talking about preventing credential leak by physical access as discussed in #2886. 

It is OK to store them cleartext on the device. But you should never ever make them readable from the web application or from any other network access.

**Comment by caco3** [comment]
I agree that it is not good practice. How ever @jomjol (the core dev) once decided he does not want to have the responsibility to protect the device (see eg. https://github.com/jomjol/AI-on-the-edge-device/issues/2886). There are several reasons for this. One is the extra work for the devs and the skills that are needed to make it really right and save. An other reason is the lack of CPU resource for this. The ESP simply is very limited and we already are on our limits RAM wise.

**Comment by caco3** [comment]
>  files containing credentials write only

Yes, I think that is the only feasable way unless we restrict the access to the UI.

**Comment by caco3** [comment]
There is now basic authentication available for the Web UI which can be used (although it is not a good solution it is better than nothing).
